src/tree/v2/ResourceTreeDataProviderBase.ts

```diff
@@ -65,7 +66,10 @@ export abstract class ResourceTreeDataProviderBase extends vscode.Disposable imp
 
     async getTreeItem(element: ResourceGroupsItem): Promise<vscode.TreeItem> {
         const treeItem = await element.getTreeItem();
-        treeItem.id = this.itemCache.getId(element);
+
+        if (!(element instanceof BranchDataProviderItem)) {
+            treeItem.id = this.itemCache.getId(element);
```

Comment:
If the returned tree item already has an `id`, should we abide by it (rather than override it)?